Microsoft.Toolkit.Uwp.UI.Controls/Graph/SharePointFiles/SharePointFiles.cs

```diff
@@ -0,0 +1,558 @@
+            _cancel = GetTemplateChild("cancel") as Button;
+            _hasMore = GetTemplateChild("hasMore") as Button;
+            _thumbnail = GetTemplateChild("thumbnail") as Grid;
+            _thumbnailImage = GetTemplateChild("thumbnailImage") as Windows.UI.Xaml.Controls.Image;
+            _details = GetTemplateChild("details") as ScrollViewer;
+            _status = GetTemplateChild("status") as TextBlock;
+            _list.SelectionChanged += List_SelectionChanged;
+            _list.ItemClick += List_ItemClick;
+            _back.Click += Back_Click;
+            _upload.Click += Upload_Click;
+            _share.Click += Share_Click;
+            _download.Click += Download_Click;
+            _delete.Click += Delete_Click;
+            _error.Click += Error_Click;
+            _cancel.Click += Cancel_Click;
+            _hasMore.Click += LoadMore_Click;
+
+            base.OnApplyTemplate();
+        }
+
+        private async void List_ItemClick(object sender, ItemClickEventArgs e)
+        {
+            if (e.ClickedItem is DriveItem driveItem && driveItem.Folder != null)
+            {
+                _driveItemPath.Push(driveItem.Id);
+                _back.Visibility = Visibility.Visible;
+                await LoadFiles(driveItem.Id);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves an appropriate Drive URL from a SharePoint document library root URL
+        /// </summary>
+        /// <param name="rawDocLibUrl">Raw URL for SharePoint document library</param>
+        /// <returns>Drive URL</returns>
+        public async Task<string> GetDriveUrlFromSharePointUrl(string rawDocLibUrl)
+        {
+            if (string.IsNullOrEmpty(rawDocLibUrl))
+            {
+                return rawDocLibUrl;
+            }
+
+            rawDocLibUrl = WebUtility.UrlDecode(rawDocLibUrl);
+
+            Match match = Regex.Match(rawDocLibUrl, @"(https?://([^/]+)((/[^/?]+)*?)(/[^/?]+))(/(Forms/\w+.aspx)?)?(\?.*)?$", RegexOptions.IgnoreCase);
+            string docLibUrl = match.Groups[1].Value;
+            string hostName = match.Groups[2].Value;
+            string siteRelativePath = match.Groups[3].Value;
+            if (string.IsNullOrEmpty(siteRelativePath))
+            {
+                siteRelativePath = "/";
+            }
+
+            Site site = await _graphClient.Sites.GetByPath(siteRelativePath, hostName).Request().GetAsync();
+            ISiteDrivesCollectionPage drives = await _graphClient.Sites[site.Id].Drives.Request().GetAsync();
+
+            Drive drive = drives.SingleOrDefault(o => WebUtility.UrlDecode(o.WebUrl).Equals(docLibUrl, StringComparison.CurrentCultureIgnoreCase));
+            if (drive == null)
+            {
+                throw new Exception("Drive not found");
+            }
+
+            return _graphClient.Drives[drive.Id].RequestUrl;
+        }
+
+        private async Task InitDrive(string driveUrl)
```

Comment:
Rename by `InitDriveAsync` for consistency

Reply:
@Odonno done